handlers/aws/utils.py

```diff
@@ -2,30 +2,47 @@
 # or more contributor license agreements. Licensed under the Elastic License 2.0;
 # you may not use this file except in compliance with the Elastic License 2.0.
 
+from typing import Any
+
 _available_triggers: dict[str, str] = {"aws:sqs": "sqs"}
 
 
-def _enrich_event(event: dict[str, any], event_type: str, dataset: str, namespace: str):
-    event["data_stream"] = {
-        "type": event_type,
-        "dataset": dataset,
-        "namespace": namespace,
-    }
+def from_s3_uri_to_bucket_name_and_object_key(s3_uri: str) -> tuple[str, str]:
+    if not s3_uri.startswith("s3://"):
+        raise ValueError(f"Invalid s3 uri provided: {s3_uri}")
+
+    s3_uri = s3_uri.strip("s3://")
 
-    event["event"] = {"dataset": dataset, "original": event["fields"]["message"]}
+    bucket_name_and_object_key = s3_uri.split("/", 1)
```

Comment:
Nit. Use urllib to parse urls? https://docs.python.org/3/library/urllib.parse.html